### PR TITLE
Fix UI Culture parsing bug

### LINKED ISSUE #1872: Culture and UI culture are not applied correctly in Blazor application
When using the .NET cookie (.AspNetCore.Culture) to set culture and UI culture of a Blazor application (as described here https://learn.microsoft.com/en-us/aspnet/core/blazor/globalization-localization?view=aspnetcore-8.0#dynamically-set-the-server-side-culture-by-user-preference), the culture (i.e., numbers/dates formatting) and the UI culture (i.e., language) are not set correctly after the introduction of the Azure SignalR service. Examples:

Without Azure Signal R (expected behavior):
- When cookie value is "c=en|uic=it", english is used for numbers/dates formatting, while italian is used for texts
- When cookie value is "c=it|uic=en, italian is used for numbers/dates formatting,  while english is used for texts

After the introduction of Azure Signal R (with .AddAzureSignalR())
- When cookie value is "c=en|uic=it", english is used for both numbers/dates formatting and texts
- When cookie value is "c=it|uic=en, italian is used for both numbers/dates formatting and texts

-> it seems that with Azure SignalR, only the "c" part of the cookie is used to set both culture and UI culture of the application.

### Further technical details
- .NET version: 7.0.403
- Azure SignalR SDK version: 1.22.0

### PR DESCRIPTION
**Pull request by xingsy97**
Fix Bug #1872
Root cause: https://github.com/Azure/azure-signalr/issues/1872#issuecomment-1839163024

### CODE REVIEW

File: src/Microsoft.Azure.SignalR/HubHost/NegotiateHandler.cs [hunk 1]
```diff
@@ -80,6 +80,7 @@ public async Task<NegotiationResponse> Process(HttpContext context)
             var claims = BuildClaims(context);
             var request = context.Request;
             var cultureName = context.Features.Get<IRequestCultureFeature>()?.RequestCulture.Culture.Name;
+            var uiCultureName = context.Features.Get<IRequestCultureFeature>()?.RequestCulture.UICulture.Name;
```
Comment (JialinXin) [inline src/Microsoft.Azure.SignalR/HubHost/NegotiateHandler.cs:83]: Also add UT?

File: src/Microsoft.Azure.SignalR.Common/Constants.cs [hunk 2]
```diff
@@ -77,6 +77,7 @@ public static class QueryParameter
             public const string OriginalPath = "asrs.op";
             public const string ConnectionRequestId = "asrs_request_id";
             public const string RequestCulture = "asrs_lang";
+            public const string RequestUiCulture = "asrs_ui_lang";
```
Comment (Y-Sindo) [inline src/Microsoft.Azure.SignalR.Common/Constants.cs:80]: Do we need changes in the runtime?

File: src/Microsoft.Azure.SignalR/HubHost/NegotiateHandler.cs (hunk 1 above)
Comment (xingsy97) [reply src/Microsoft.Azure.SignalR/HubHost/NegotiateHandler.cs:83]: added

File: src/Microsoft.Azure.SignalR.Common/Constants.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (xingsy97) [reply src/Microsoft.Azure.SignalR.Common/Constants.cs]: updated

**Review by vicancy — APPROVED** [review]

**Comment by xingsy97** [comment]
I cannot push more commit to this pr for it's not created on my own repo. Will open a new one.

**Pull request closed without merging**